askama_shared/src/generator.rs

```diff
@@ -1531,17 +1515,21 @@ impl<'a, S: std::hash::BuildHasher> Generator<'a, S> {
         DisplayWrap::Unwrapped
     }
 
-    fn visit_target(&mut self, buf: &mut Buffer, target: &Target<'a>) {
+    fn visit_target(&mut self, buf: &mut Buffer, initialized: bool, target: &Target<'a>) {
         match target {
             Target::Name(name) => {
                 let name = normalize_identifier(name);
-                self.locals.insert_with_default(name);
+                if initialized {
+                    self.locals.insert(name, LocalMeta::initialized());
+                } else {
+                    self.locals.insert_with_default(name);
+                }
```

Comment:
Here's an example where I'd probably write a `match` instead. What do you think?